### **Transcript quantification**
#### <u>plate-based protocols (full length)</u>
an even coverage of transcripts is not always achieved with full-length protocols and therefore specific regions across the gene body may still be biased. full-length protocols is that they allow for the detection of splice variants.

#### <u>Drop/Tag-based</u>
it can mitigate the bias of transcript amplification. It can not align reads to the distinguishing transcripts. In microfluidic based protocols only about 10% of the transcripts of the cell are recovered, but it is sufficient to identify the cell types. In the microfluidic device-based methods (Drop-seq, Indrop, 10x), reads originate from valid barcodes was 75% for 10x, while the others are less than 30%. But 10X genomics favored the capture and amplification of shorter genes and genes with higher GC content.

### **scRNA vs snRNA**
snRNA-seq could not always capture all cell types because of the vulnerable cell types such as glutamatergic neurons in the brain. Besides, the snRNA-seq rely on the fresh tissue. snRNA-seq does not need the cell dissociation and are more resistant to mechanical force, and safe to the frozen tissue. It should be noticed that the cell dissociation has a strong effect on the potentially observable cell types.

Mapping to the genome
Because cell ranger use the entire genome for reference, the reads mapping outside the annotated transcripts axons can still be accounted for.


